packages/web-media-box/src/hls-parser/tags/tagWithAttributesProcessors.ts

```diff
@@ -492,14 +494,44 @@ export class ExtXPreloadHint extends TagWithAttributesProcessor {
   protected readonly tag = EXT_X_PRELOAD_HINT;
 
   protected safeProcess(tagAttributes: Record<string, string>, playlist: ParsedPlaylist): void {
-    const preloadHint = {
-      type: tagAttributes[ExtXPreloadHint.TYPE] as HintType,
-      uri: tagAttributes[ExtXPreloadHint.URI],
-      byterangeStart: Number(tagAttributes[ExtXPreloadHint.BYTERANGE_START]),
-      byterangeLength: Number(tagAttributes[ExtXPreloadHint.BYTERANGE_LENGTH]),
-    };
+    const type = tagAttributes[ExtXPreloadHint.TYPE] as PreloadHintType;
+    const uri = tagAttributes[ExtXPreloadHint.URI];
+    const pStart = tagAttributes[ExtXPreloadHint.BYTERANGE_START];
+    const pLength = tagAttributes[ExtXPreloadHint.BYTERANGE_LENGTH];
+
+    /**
+     * There are 4 scenarios with Byte Range for preload-hint
+     * 1. Start is available, Length is available:
+     * Request resource from start till (start + length - 1)
+     * 2. Start is available, Length is not available:
+     * Request resource from start till the end of the resource
+     * 3. Start is not available, Length is available:
+     * Request from 0 till (length - 1)
+     * 4. Start is not available, Length is not available:
+     * Request entire resource (default scenario)
+     */
+
+    let byteRange;
+
+    if (pStart && pLength) {
+      const start = Number(pStart);
+      const end = start + Number(pLength) - 1;
+      byteRange = { start, end };
+    } else if (pStart && !pLength) {
+      byteRange = { start: Number(pStart), end: Number.MAX_SAFE_INTEGER };
+    } else if (!pStart && pLength) {
+      byteRange = { start: 0, end: Number(pLength) - 1 };
+    }
 
-    playlist.preloadHints.push(preloadHint);
+    const preloadHint = { uri, byteRange };
```

Comment:
For my own understanding, we're OK with byteRange in `preloadHint` being `undefined` in the default scenario?

Reply:
yes, this is expected!